tests/test_e2e_41_kytos_auth.py

```diff
@@ -0,0 +1,116 @@
+import os
+import json
+import time
+import base64
+import shutil
+import requests
+from tests.helpers import NetworkTest
+from kytos.core.auth import UserController
+
+CONTROLLER = '127.0.0.1'
+KYTOS_API = 'http://%s:8181/api/kytos' % CONTROLLER
+
+
+class TestE2EKytosAuth:
+    net = None
+
+    def setup_method(self, method):
+        """
+        It is called at the beginning of every class method execution
+        """
+        self.net.start_controller(clean_config=True, enable_all=False)
+        self.user_data = {
+            "username": os.environ.get("USER_USERNAME"),
+            "password": os.environ.get("USER_PASSWORD"),
+            "email": os.environ.get("USER_EMAIL")
+        }
+        self.user_controller = UserController()
+        self.user_controller.create_user(self.user_data)
+        self.token = self.get_token()
+        time.sleep(5)
+
+    @classmethod
+    def setup_class(cls):
+        cls.net = NetworkTest(CONTROLLER)
+        cls.net.start()
+        # rotate logfile (copy/truncate strategy)
+        logfile = '/var/log/syslog'
+        shutil.copy(logfile, logfile + '-' + time.strftime("%Y%m%d%H%M%S"))
+        open(logfile, 'w').close()
+
+    @classmethod
+    def teardown_class(cls):
+        cls.net.stop()
+
+    def get_token(self):
+        api_url = KYTOS_API + '/core/auth/login/'
+        username = os.environ.get("USER_USERNAME")
+        password = os.environ.get("USER_PASSWORD")
+
+        header = {
+            "Authorization": "Basic "
+            + base64.b64encode(
+                bytes(username + ":" + password, "ascii")
+            ).decode("ascii")
+        }
+        answer = requests.get(api_url, headers=header, auth=(username, password))
+
+        return answer.json()['token']
+
+
+    def test_authenticate_user(self):
+        api_url = KYTOS_API + '/core/auth/login/'
+        username = os.environ.get("USER_USERNAME")
+        password = os.environ.get("USER_PASSWORD")
+        header = {
+            "Authorization": "Basic "
+            + base64.b64encode(
+                bytes(username + ":" + password, "ascii")
+            ).decode("ascii")
+        }
+        answer = requests.get(api_url, headers=header, auth=(username, password))
+        assert answer.status_code == 200
```

Comment:
Looks good, @Alopalao. This is the happy path of this endpoint, let's also cover the fail case since in practice, users will eventually insert wrong credentials and we need to make sure that from e2e perspective that also works. You can use this same test to make another GET request but passing wrong credentials and asserting the status code and error message.

Reply:
Looks like I cannot get the error message when `password` is incorrect. It raises `Unauthorized` which is later caught to execute this (from [kytos/core/auth.py](https://github.com/Alopalao/kytos/blob/faa37ec2764a2ad897cdda57bb54c6e7e2bef7c1/kytos/core/auth.py#L292-L294)):
```
except Unauthorized:
            result = "Incorrect password"
            return result, HTTPStatus.UNAUTHORIZED.value
```
I think it would be better to delete the try/except codes and just raise `Unauthorized("Incorrect password")`

Reply:
Oh, got it.